### ISSUE TITLE
List Reporters cause large sb3 file size

### ISSUE THREAD

**Issue by griffpatch** [LLK/scratch-vm#2294]
### Actual Behavior

I have been reaching the limit of file size when trying to save my projects in scratch. I discovered that deleting all the contents of my lists helped, but not always. This led me to nosey around in the sb3 file, and I discovered that if you have viewed a large list (list reporter) in your project. Then hide it again BEFORE deleting all the contents, the content is persisted in the sb3 file even after the actual list is emptied. It appears the list reporters remember their previous content when hidden and this is saved to the sb3 file even though lists may have changed since.

p.s. 200,000 item maximum list size is a little too small :( - What chance is there to raise the limit?

### Steps to Reproduce

1. Create a large list
2. View the list (as a list reporter)
3. Hde the list
4. Delete all rows in the list
5. Export your sb3 file and inspect the contents
6. Gasp as you notice all your list content is still there!

### Operating System and Browser

Chrome on Windows PC

**Comment by qucchia** [comment]
This is true: it means that the contents of the list is actually saved in two places, making it reach the file size limit more easily.

It is possible to pass the limit of 200,000 items by:
1. Manually adding items by clicking on the list and pressing enter.
2. Importing a text file that has more than 200,000 lines.
3. Importing an SB3 or SPRITE3 file that has a list with over 200,000 items.
Unfortunately this means that you must prepare the list before it's being played, and the script will have to replace the item you need. It also means the length of the list won't be able to change.

**Comment by AmazingMech2418** [comment]
> It is possible to pass the limit of 200,000 items by:
> 
> 1. Manually adding items by clicking on the list and pressing enter.
> 2. Importing a text file that has more than 200,000 lines.
> 3. Importing an SB3 or SPRITE3 file that has a list with over 200,000 items.
>    Unfortunately this means that you must prepare the list before it's being played, and the script will have to replace the item you need. It also means the length of the list won't be able to change.

These are possible, but dynamically adding to the list to make it larger than 200,000 items is not. Though, unfortunately, it does not seem like the list limit could be increased since it could cause the tab to run out of memory or might exceed the project storage limits. However, it could be possible that lists be greater than this length dynamically, but that the lists are not saved in the SB3 file--only locally--and that you could choose an option for this. Like maybe when creating a list, have another checkbox that is something like "Large list (doesn't save with project)" or something, and if you check that, it saves this as a property of the list, meaning that it doesn't make it so the project won't save.
